core/object/object.cpp

```diff
@@ -493,7 +493,7 @@ void Object::get_property_list(List<PropertyInfo> *p_list, bool p_reversed) cons
 		uint32_t pcount;
 		const GDNativePropertyInfo *pinfo = _extension->get_property_list(_extension_instance, &pcount);
 		for (uint32_t i = 0; i < pcount; i++) {
-			p_list->push_back(PropertyInfo(Variant::Type(pinfo[i].type), pinfo[i].class_name, PropertyHint(pinfo[i].hint), pinfo[i].hint_string, pinfo[i].usage, pinfo[i].class_name));
+			p_list->push_back(PropertyInfo(pinfo[i]));
```

Comment:
There's a logic difference here (and likewise in `script_language_extension.cpp`'s replacements) if `pinfo[i].name` is not the same as `pinfo[i].class_name`. In all three cases the multi-arg constructor was assigning `class_name` to the new PropertyInfo's `name`.

Reply:
Ah I see you pointed this out in the PR description. @reduz should likely check if this was intentional or indeed an oversight.